web/src/components/FilingsCalendar.test.tsx

```diff
@@ -140,6 +142,99 @@ describe("<FilingsCalendar />", () => {
+      const annualReport = generateTaxFiling({
+        identifier: "annual-report",
+        dueDate: dueDate.format("YYYY-MM-DD"),
+      });
+
+      const userData = generateUserData({
+        profileData: generateProfileData({
+          legalStructureId: randomLegalStructure(true).id,
+          operatingPhase: randomElementFromArray(
+            OperatingPhases.filter((obj) => obj.displayTaxAccessButton === true)
+          ).id,
+        }),
+        taxFilingData: generateTaxFilingData({ filings: [annualReport] }),
+      });
+
+      const operateReferences: Record<string, OperateReference> = {
+        "annual-report": {
+          name: "Annual Report",
+          urlSlug: "annual-report-url",
+          urlPath: "annual_report-url-path",
+        },
+      };
+
+      renderFilingsCalendar(operateReferences, userData);
+
+      expect(screen.getByTestId("get-tax-access")).toBeInTheDocument();
+    });
+
+    it("hides button on filings calendar when sp/gp", () => {
+      const whateverReport = generateTaxFiling({
+        identifier: "whatever-report",
+      });
+      const userData = generateUserData({
+        profileData: generateProfileData({
+          legalStructureId: randomLegalStructure(false).id,
+          operatingPhase: randomElementFromArray(
+            OperatingPhases.filter((obj) => obj.displayTaxAccessButton === true)
+          ).id,
+        }),
+        taxFilingData: generateTaxFilingData({ filings: [whateverReport] }),
+      });
+
+      const operateReferences: Record<string, OperateReference> = {
+        "whatever-report": {
+          name: "Whatever",
+          urlSlug: "whatever-report-url",
+          urlPath: "whatever_report-url-path",
+        },
+      };
+
+      renderFilingsCalendar(operateReferences, userData);
+
+      expect(screen.queryByTestId("get-tax-access")).not.toBeInTheDocument();
+    });
+
+    it("hides button on filings calendar when displayTaxAccessButton is false", () => {
+      const dueDate = getCurrentDate().add(2, "months");
+      const annualReport = generateTaxFiling({
+        identifier: "annual-report",
+        dueDate: dueDate.format("YYYY-MM-DD"),
+      });
+
+      const userData = generateUserData({
+        profileData: generateProfileData({
+          legalStructureId: randomLegalStructure(true).id,
+          operatingPhase: randomElementFromArray(
+            OperatingPhases.filter((obj) => obj.displayTaxAccessButton !== true)
+          ).id,
+        }),
+        taxFilingData: generateTaxFilingData({ filings: [annualReport] }),
+      });
+
+      const operateReferences: Record<string, OperateReference> = {
+        "annual-report": {
+          name: "Annual Report",
+          urlSlug: "annual-report-url",
+          urlPath: "annual_report-url-path",
+        },
+      };
+
+      renderFilingsCalendar(operateReferences, userData);
+
+      expect(screen.queryByTestId("get-tax-access")).not.toBeInTheDocument();
+    });
+  });
+
```

Comment:
at first, I was confused by these tests here because this is the FilingsCalendar test and there is no logic in the FilingsCalendar component that handles whether to show/hide this button the way these tests describe. 

Looking further, I discovered the logic that actually controls this lives in the TaxAccess component.  Remembering our techtro conversation with Faraz a few weeks ago, I think that logic doesn't belong at the component level. I believe we agreed that child components should not control whether or not they exist. Instead, parent components should decide whether or not to render that child on the page, and the child should just exist, just do its thing without any logic where it just doesn't exist in certain conditions.

All that to say, these tests are correctly placed but the logic should be moved into the FilingsCalendar component

Reply:
I would usually agree, but this component is being rendered in multiple locations with identical logic, I think we need to refactor all the calendar stuff, but I didn't really want to do that at this moment. Maybe I should just duplicate it in both locations thoug